src/datetime.js

```diff
@@ -1844,33 +1831,42 @@ export default class DateTime {
     suppressMilliseconds = false,
     includeOffset = true,
     extendedZone = false,
+    precision = "milliseconds",
   } = {}) {
-    if (!this.isValid) {
-      return null;
-    }
-
-    const ext = format === "extended";
+    if (!this.isValid) return null;
 
-    let c = toISODate(this, ext);
-    c += "T";
-    c += toISOTime(this, ext, suppressSeconds, suppressMilliseconds, includeOffset, extendedZone);
+    const isExtendedFormat = format === "extended",
+      precisionUnitIndex = orderedUnits.indexOf(normalizeUnit(precision));
+    if (precisionUnitIndex < 0) throw new InvalidUnitError(precision);
+
+    let c = toISODate(this, isExtendedFormat, precisionUnitIndex);
+    c += precisionUnitIndex >= 3 ? "T" : ""; //only show T if time is visible
+    c += toISOTime(
+      this,
+      isExtendedFormat,
+      suppressSeconds,
+      suppressMilliseconds,
+      includeOffset,
+      extendedZone,
+      precisionUnitIndex
+    );
     return c;
   }
 
   /**
    * Returns an ISO 8601-compliant string representation of this DateTime's date component
    * @param {Object} opts - options
    * @param {string} [opts.format='extended'] - choose between the basic and extended format
+   * @param {string} [opts.precision='day'] - truncate output to desired precision: 'years', 'months', or 'days'.
    * @example DateTime.utc(1982, 5, 25).toISODate() //=> '1982-05-25'
    * @example DateTime.utc(1982, 5, 25).toISODate({ format: 'basic' }) //=> '19820525'
+   * @example DateTime.utc(1982, 5, 25).toISODate({ precision: 'month' }) //=> '1982-05'
    * @return {string|null}
    */
-  toISODate({ format = "extended" } = {}) {
-    if (!this.isValid) {
-      return null;
-    }
-
-    return toISODate(this, format === "extended");
+  toISODate({ format = "extended", precision = "day" } = {}) {
+    if (!this.isValid) return null;
+    const precisionUnitIndex = orderedUnits.indexOf(normalizeUnit(precision));
+    return toISODate(this, format === "extended", precisionUnitIndex);
```

Comment:
add `if (precisionUnitIndex < 0) throw new InvalidUnitError(precision)` here... except somehow the test is already passing, so investigate why that test is either not working, or else why the system is somehow throwing the correct error anyway.

Reply:
the error throwing was being handled by the `normalizeUnit` method, so I removed the superfluous error throwing logic.